app/com/linkedin/drelephant/mapreduce/MapReduceFetcherHadoop2.java

```diff
@@ -123,13 +125,9 @@ public MapReduceApplicationData fetchData(AnalyticJob analyticJob) throws IOExce
         }
         jobData.setDiagnosticInfo(diagnosticInfo);
       } else {
-        // Should not reach here
-        throw new RuntimeException("Job state not supported. Should be either SUCCEEDED or FAILED");
-      }
-    } finally {
-      ThreadContextMR2.updateAuthToken();
```

Comment:
Shouldn't this be here again now? I think you probably need to re-add all the auth stuff, right?